libparsers/tests/xml/route_marker.cpp

```diff
@@ -0,0 +1,212 @@
+        "<?xml version=\"1.0\" encoding=\"utf-8\"?>"
+        "<openBVE xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\">"
+            "<ImageMarker>"
+                    "<Early>"
+                        "<Time>A</Time>"
+                        "<Image>Cloudy.png</Image>"
+                    "</Early>"
+                    "<OnTime>"
+                        "<Image>Sunny.png</Image>"
+                    "</OnTime>"
+                    "<Late>"
+                        "<Time>A</Time>"
+                        "<Image>Foggy.png</Image>"
+                    "</Late>"
+                    "<Distance>-1</Distance>"
+                    "<Timeout>-1</Timeout>"
+                    "<trains></trains>"
+            "</ImageMarker>"
+        "</openBVE>"s;
+    // clang-format off
+    auto rel_file_func = [](std::string base, std::string rel){ return base + "/" + rel;}; 
+	parsers::errors::multi_error_t output_errors;
+    auto const output = rm::parse("some_file.xml"s, image_based_marker, output_errors,rel_file_func);
+	REQUIRE(output.is<rm::image_marker>());
+	auto& image_marker =
+	    output.get_unchecked<rm::image_marker>();
+	CHECK_EQ(image_marker.early_time, 0);
+    CHECK(!output_errors.empty());
+    CHECK_EQ(output_errors["some_file.xml"].size(), 2);
+    CHECK(!image_marker.using_early);
+    CHECK(image_marker.using_ontime);
+    CHECK(!image_marker.using_late);
+
+}
+
+TEST_CASE("libparsers - xml - router_marker - early/late nodes reuqire time and image nodes, ontime nodes require a image node"){
+    // clang-format off
+    std::string const image_based_marker =
+            "<?xml version=\"1.0\" encoding=\"utf-8\"?>"
+            "<openBVE xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\">"
+                "<ImageMarker>"
+                    "<Early>"
+                    // No time or image nodes.
+                    "</Early>"
+                    "<OnTime>"
+                    // No nodes.
+                    "</OnTime>"
+                    "<Late>"
+                    // No time or image nodes
+                    "</Late>"
+                    "<Distance>-1</Distance>"
+                    "<Timeout>-1</Timeout>"
+                    "<trains></trains>"
+                "</ImageMarker>"
+            "</openBVE>"s;
+    // clang-format off
+    auto rel_file_func = [](std::string base, std::string rel){ return base + "/" + rel;};
+    parsers::errors::multi_error_t output_errors;
+    auto const output = rm::parse("some_file.xml"s, image_based_marker, output_errors,rel_file_func);
+            REQUIRE(output.is<rm::image_marker>());
+    auto& image_marker =
+            output.get_unchecked<rm::image_marker>();
+    CHECK(!output_errors.empty());
+    CHECK_EQ(output_errors["some_file.xml"].size(), 3);
+    CHECK_EQ(image_marker.early_time, 0);
+    CHECK_EQ(image_marker.early_filename,"some_file.xml/"s);
```

Comment:
You shouldn't even check those. By definition what they are is irrelevant because of the booleans.